src/fhirtypes/StructureDefinition.ts

```diff
@@ -82,6 +83,21 @@ export class StructureDefinition {
    */
   private _sdStructureDefinition: StructureDefinition;
 
+  validate(): ValidationError[] {
+    const validationErrors: ValidationError[] = [];
+    this.elements.forEach(e => {
+      e.validate().forEach(err => {
+        validationErrors.push(
+          new ValidationError(
+            err.issue,
+            `${e.diffId().replace(/(\S):(\S+)/g, (_, c1, c2) => `${c1}[${c2}]`)} ^${err.fshPath}`
```

Comment:
Somehow I've never seen this style of `replace` call before, but it's pretty nice for quick simple regexes with capture groups.